File: 5_kyu/Human_Readable_Time.lua
local solution = {}

-- Formats time seconds in "hh:mm:ss" with
-- ss = seconds, mm = minutes, hh = hours
-- seconds must be in range 0..99:59:59
function solution.make_readable(seconds)
  if seconds < 0 then
    return "ERR <0"
  elseif seconds >= 100*60*60 then
    return "ERR >100"
  end
  local h,m,s
  s = seconds % 60
  m = seconds // 60
  h = m // 60
  m = m % 60
  return string.format("%02d:%02d:%02d",h,m,s)
end

return solution

File: 5_kyu/Human_Readable_Time.rs
// Format time as HH:MM:SS.
fn make_readable(seconds: u32) -> String {
    let ss = seconds % 60;
    let minutes = seconds / 60;
    let mm = minutes % 60;
    let hh = minutes / 60;
    format!("{hh:02}:{mm:02}:{ss:02}")
}
